modules/workflow-service-impl/src/main/java/org/opencastproject/workflow/impl/WorkflowServiceImpl.java

```diff
@@ -2394,71 +2390,44 @@ public synchronized void cleanupWorkflowInstances(int buffer, WorkflowState stat
 
   @Override
   public void repopulate(final String indexName) throws Exception {
-    List<Job> jobs = new ArrayList<>();
-    try {
-      for (Job job : serviceRegistry.getJobs(WorkflowService.JOB_TYPE, null)) {
-        if (WorkflowServiceImpl.Operation.START_WORKFLOW.toString().equals(job.getOperation())) {
-          jobs.add(job);
-        }
-      }
-    } catch (ServiceRegistryException e) {
-      logger.error("Unable to load the workflows jobs: {}", e.getMessage());
-      throw new ServiceException(e.getMessage());
-    }
-
-    final String destinationId = WorkflowItem.WORKFLOW_QUEUE_PREFIX + WordUtils.capitalize(indexName);
-    if (jobs.size() > 0) {
-      logger.info("Populating index '{}' with {} workflows", indexName, jobs.size());
-      final int total = jobs.size();
+    List<String> workflows =  serviceRegistry.getJobPayloads(Operation.START_WORKFLOW.toString());
```

Comment:
Nitpick: Shouldn't this variable be named payloads?  Alternatively, we should rename the method.

Reply:
´getJobPayloads(…)` will return the payload of any type of job. It just happens to be that the payload of a `START_WORKFLOW` job is a serialized workflow instance. That's why the name `workflows` for the variable does make sense here.